kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md

```diff
@@ -81,41 +53,10 @@ module EVM-OPTIMIZATIONS [kore]
       </ethereum>
       ...
     </kevm>
-    requires ( Gbase < SCHED > <=Gas GAVAIL )
+    requires ( notBool USEGAS orBool Gbase < SCHED > <=Gas GAVAIL )
```

Comment:
Any reason to use `notBool ... orBool ...` here instead of `#if ... #then ... #else ... #fi`? Given that we'll have direct implementation of `#ite` in the backend, that will likely perform better, as it will get directly reduced to one branch or the other and won't have to evaluate both sides of the `orBool`.

Reply:
Given `domains.md`, the RHS of `orBool` will be evaluated only when the LHS evaluates to false, so the only extra computation will be that of `notBool USEGAS`, for a concrete `USEGAS: true`. Happy to revert this to `#if`.

Reply:
I think we should use the `#ite` specifically because it's _hooked_ and we can guarantee the evaluation order of it with the backend (making sure to inspect the boolean condition before evaluating the arguments of each branch).

I'm not talking about the k-rules, I'm talking about how the backend simplifies terms. It does not pay attention to left/right, instead it decides bottom-up or top-down, and no guarantee on the order of arguments that it evaluates. Even with the k-rules which say things like `rule true orBool _ => true`, the backend can still evaluate the second argument before firing this rule.

Let's use `#ite`.

Reply:
Yes, this is what's happening now.